utils/s3/retryable_http_client.cc

```diff
@@ -56,14 +56,24 @@ future<> retryable_http_client::do_retryable_request(http::request req, http::ex
     }
 }
 
+future<> retryable_http_client::make_request(http::request req, std::optional<http::reply::status_type> expected, abort_source* as) {
+    co_return co_await make_request(
+        std::move(req),
+        [](const http::reply& rep, input_stream<char>&& in_) -> future<> {
```

Comment:
There's `s3::ignore_reply()` helper in utils/s3/client.cc that callers of client::make_request use for their convenience. Why adding an heavyweight overload to achieve the same?